### ISSUE TITLE
Fix remaining issues reported by golint

### ISSUE THREAD

**Issue by thomasmodeneis** [ChainSafe/gossamer#586]
After #559 is closed and #574 is merged, the following linter skip steps should be removed and a new PR should fix all the reported issues.

Special case for `"which can be annoying to use"`, this one may require a refactor on exported functions that are returning a non exported type `node`, the golint recommendation is to avoid doing it. The other special case `"and that stutters; consider calling this"` would also require special attention and decision.

```   
    - linters:
        - golint
      text: "which can be annoying to use"

    - linters:
        - golint
      text: "and that stutters; consider calling this"

    - linters:
        - golint
      text: "should have comment or be unexported"

    - linters:
        - golint
      text: "or a comment on this block"

    - linters:
        - golint
      text: "error strings should not be capitalized or end with punctuation or a newline"
```

## Steps to Reproduce (for bugs)

1.
2.
3.
4.

## Your Environment
* Version used:
* Commit hash:
* Go Version:
* Operating System and version (include distro if Linux):
* Link to your project or reproducible gist:
* I intend to submit a pull request to fix this issue: [ ]
